query-connector/src/app/tests/integration/user_management.test.ts

```diff
@@ -48,12 +56,15 @@ describe("User Management Integration Tests", () => {
   /**
    * Tests adding a new user if they do not already exist.
    */
-  test("should add a user if they do not exist", async () => {
-    const result = await addUserIfNotExists(TEST_USER);
+  test.each([
+    { user: TEST_SUPER_USER, role: RoleTypeValues.SuperAdmin },
+    { user: TEST_USER, role: RoleTypeValues.Standard },
+  ])("should add a user if they do not exist", async ({ user, role }) => {
+    const result = await addUserIfNotExists(user);
```

Comment:
feel free to do this in a followup!

it looks like this function returns the user if they do exist. from where i'm reading, we might want to assert that they don't exist first if we're trying to test the state where they get added if they don't exist. in that case, may also want to do some cleanup after to delete the newly added user.